Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbNativeWindow.cpp

```diff
@@ -219,17 +231,143 @@ namespace AzFramework
         }
     }
 
+
+    // This function will draw the splash image if one is set
+    // otherwise it will return without doing anything
+    void XcbNativeWindow::DrawSplash()
+    {
+#ifndef SPLASH_IMAGE_EXISTS
+    return;  // failed to find splash image no need to draw
+#else
```

Comment:
* You can simply use `#if` instead of `ifdef/ifndef`. This works for situations where `SPLASH_IMAGE_EXISTS` is defined or not, and if its defined, it will evaluate the value (true/false or 1/0)
* Seems unnecessary for both the `return` statement and the inverted #ifndef block here. The implementation of this function could simply be wrapped around a `#if SPLASH_IMAGE_EXISTS`/`#endif` block

Reply:
Oh you are right this makes a lot of sense